description: use the correct program counter data type for arm64
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/ntdll/signal_arm64.c
+++ b/dlls/ntdll/signal_arm64.c
@@ -611,9 +611,8 @@ static NTSTATUS raise_exception( EXCEPTI
     return STATUS_SUCCESS;
 }
 
-static inline DWORD is_write_fault( DWORD *pc )
+static inline DWORD is_write_fault( DWORD64 inst )
 {
-    DWORD inst = *pc;
     if ((inst & 0xbfff0000) == 0x0c000000   /* C3.3.1 */ ||
         (inst & 0xbfe00000) == 0x0c800000   /* C3.3.2 */ ||
         (inst & 0xbfdf0000) == 0x0d000000   /* C3.3.3 */ ||
@@ -637,7 +636,7 @@ static void segv_handler( int signal, si
 {
     EXCEPTION_RECORD *rec;
     ucontext_t *context = ucontext;
-    DWORD *orig_pc = PC_sig(context);
+    DWORD64 orig_pc = PC_sig(context);
 
     /* check for page fault inside the thread stack */
     if (signal == SIGSEGV &&
